File: src/contests/cg/puzzle/swift/euclids-algorithm.swift
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

let parts = readLine()!.split(separator: " ").map { Int($0)! }
let a = parts[0], b = parts[1]
var x = a, y = b
while y != 0 {
    let q = x / y, r = x % y
    print("\(x)=\(y)*\(q)+\(r)")
    x = y
    y = r
}
print("GCD(\(a),\(b))=\(x)")

File: src/contests/cg/puzzle/fsharp/euclids-algorithm.fs
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

let parts = System.Console.ReadLine().Trim().Split(' ')
let a = int64 parts.[0]
let b = int64 parts.[1]
let mutable x = a
let mutable y = b
while y <> 0L do
    let q = x / y
    let r = x % y
    printfn "%d=%d*%d+%d" x y q r
    x <- y
    y <- r
printfn "GCD(%d,%d)=%d" a b x
